test/e2e-tests/support/tekton.go

```diff
@@ -0,0 +1,53 @@
+package support
+
+import (
+	v1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	"gopkg.in/yaml.v3"
+	"os"
+)
+
+func SetPipelineRunParam(name string, value v1.ParamValue, pipelineRun *v1.PipelineRun) {
+	for index := range pipelineRun.Spec.Params {
+		param := &pipelineRun.Spec.Params[index]
+		if param.Name == name {
+			param.Value = value
+		}
+	}
+}
+
+func NewStringParamValue(value string) v1.ParamValue {
+	return v1.ParamValue{
+		Type:      v1.ParamTypeString,
+		StringVal: value,
+	}
+}
+
+func NewArrayParamValue(value []string) v1.ParamValue {
+	return v1.ParamValue{
+		Type:     v1.ParamTypeArray,
+		ArrayVal: value,
+	}
+}
+
+func NewObjectParamValue(value map[string]string) v1.ParamValue {
+	return v1.ParamValue{
+		Type:      v1.ParamTypeObject,
+		ObjectVal: value,
+	}
+}
+
+func ReadFileAsPipelineRun(path string) (v1.PipelineRun, error) {
+	var pipelineRun v1.PipelineRun
+
+	bytes, err := os.ReadFile(path)
+	if err != nil {
+		return pipelineRun, err
+	}
+
+	err = yaml.Unmarshal(bytes, &pipelineRun)
+	if err != nil {
+		return pipelineRun, err
```

Comment:
>  what will the value of pipelineRun be
Because in these cases `err != nil` when the caller checks the return values from this function then there is no expectation for the value of pipelineRun

> Should nil be explicitly returned instead?
Because I am returning by value some pipelineRun value does need to be returned. I could return a default initialised one like `PipelineRun{}` or do what I did here and just return the value we currently have 

What do you think?

Reply:
Ah yes, understood. :+1: I _guess_ the value of pipelineRun is `PipelineRun{}` at this point then?